pilot/pkg/model/push_context.go

```diff
@@ -1407,7 +1407,7 @@ func (ps *PushContext) updateContext(
 func (ps *PushContext) initServiceRegistry(env *Environment, configsUpdate sets.Set[ConfigKey]) {
 	// Sort the services in order of creation.
 	allServices := SortServicesByCreationTime(env.Services())
-	if features.EnableExternalNameAlias {
+	if features.EnableExternalNameService && features.EnableExternalNameAlias {
```

Comment:
This feels unintuitive to me; why would you need to set two flags in order to get sane external name functionality. If we eventually want ExternalNameAlias to become default eventually, what's the point of a separate flag?

Reply:
EnableExternalNameAlias is already on by default which I think makes this PR less required

Reply:
Oh if that's the case then I'm a -1 on this PR. Something that's on by default shouldn't require another off-by-default flag to enable. And now that the first one has shipped, that would be a breaking change in 1.21

Reply:
This ENABLE_EXTERNAL_NAME_SERVICE is on by default so its not a breaking change.

However, there are literally 1000s of fields we read from in Kubernetes. I haven't seen a compelling argument for why *this* one needs a flag. I also think having flags of this nature in general (permanently one way without a deprecation path) are ideal.

Reply:
EnableExternalNameAlias is to solve a lot of issues caused by external name service, but for those that do not need external name service at all.  Turn on `EnableExternalNameAlias ` is costly, it is the intention. 

I think it is agreed on a knob https://github.com/istio/istio/pull/46332#issuecomment-1780378016, and i have lots of concerns on that hack way.